File: python/006.ZigzagConversion.py
class Solution:
    # @param {string} s
    # @param {integer} numRows
    # @return {string}
    def convert(self, s, numRows):
        if numRows == 1:
            return s
        dir = True
        idx = 0
        len_s = len(s)
        sub = [];
        for i in range(0,numRows):
            sub.append("")
        for i in range(0,len_s):
            sub[idx] += s[i]
            if (idx==numRows-1):
                dir = False
            elif (idx==0):
                dir = True
            if dir:
                idx += 1
            else:
                idx -= 1
        ret = ""
        for i in range(0,numRows):
            ret += sub[i]
        return ret
        

File: cpp/006.ZigzagConversion.cpp
#include <vector>

class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows==1) return s;
        vector <string> sub(numRows, "");
        string ret = "";
        int len = s.length();
        bool dir = true;
        int idx = 0;
        for (int i=0;i<len;++i)
        {
            sub[idx] += s[i];
            if (idx==numRows-1) dir = false;
            else if (idx==0) dir = true;
            if (dir) idx++;
            else idx--;
        }
        for (int i=0;i<numRows;++i)
            ret += sub[i];
        return ret;
    }
};